File: compiler-group-traslator/src/test/resources/c/If.c
#include<stdio.h>
#include<stdlib.h>
#include<stdbool.h>
#include "GC.c"
#include "ExceptionHandlingFunctions.c"
VM* vm;

int main(char* args) {
	vm = newVM();
	int a;

	a = 0 ;
	if (true) {
	}else{
	}
	assertTrue(a == 0);

	a = 0 ;
	if (true) {
		a = 1 ;
	}else{
	}
	assertTrue(a == 1);

	a = 0 ;
	if (true) {
		a = 1 ;

		a = 2 ;
	}else{
	}
	assertTrue(a == 2);

	a = 0 ;
	if (true) {
	}else{
	}
	assertTrue(a == 0);

	a = 0 ;
	if (true) {
		a = 1 ;
	}else{
	}
	assertTrue(a == 1);

	a = 0 ;
	if (true) {
		a = 1 ;
	}else{
		a = 2 ;
	}
	assertTrue(a == 1);

	a = 0 ;
	if (true) {
		a = 1 ;
	}else{
		a = 2 ;

		a = 3 ;
	}
	assertTrue(a == 1);

	a = 0 ;
	if (false) {
	}else{
	}
	assertTrue(a == 0);

	a = 0 ;
	if (false) {
		a = 1 ;
	}else{
	}
	assertTrue(a == 0);

	a = 0 ;
	if (false) {
		a = 1 ;
	}else{
	}
	assertTrue(a == 0);

	a = 0 ;
	if (false) {
		a = 1 ;
	}else{
		a = 2 ;
	}
	assertTrue(a == 2);

	a = 0 ;
	if (false) {
		a = 1 ;
	}else{
		a = 2 ;

		a = 3 ;
	}
	assertTrue(a == 3);
	return 0;
}

File: compiler-group-traslator/src/test/resources/If.java
public class If {
	public static void main(String[] args) {
		int a;

		a = 0;
		if (true) {
		}
		StaticJavaLib.assertTrue(a == 0);
		
		a = 0;
		if (true) {
			a = 1;
		}
		StaticJavaLib.assertTrue(a == 1);
		
		a = 0;
		if (true) {
			a = 1;
			a = 2;
		}
		StaticJavaLib.assertTrue(a == 2);
		
		a = 0;
		if (true) {
		} else {
		}
		StaticJavaLib.assertTrue(a == 0);
		
		a = 0;
		if (true) {
			a = 1;
		} else {
		}
		StaticJavaLib.assertTrue(a == 1);

		a = 0;
		if (true) {
			a = 1;
		} else {
			a = 2;
		}
		StaticJavaLib.assertTrue(a == 1);

		a = 0;
		if (true) {
			a = 1;
		} else {
			a = 2;
			a = 3;
		}
		StaticJavaLib.assertTrue(a == 1);
		
		a = 0;
		if (false) {
		}
		StaticJavaLib.assertTrue(a == 0);

		a = 0;
		if (false) {
			a = 1;
		}
		StaticJavaLib.assertTrue(a == 0);

		a = 0;
		if (false) {
			a = 1;
		} else {
		}
		StaticJavaLib.assertTrue(a == 0);

		a = 0;
		if (false) {
			a = 1;
		} else {
			a = 2;
		}
		StaticJavaLib.assertTrue(a == 2);

		a = 0;
		if (false) {
			a = 1;
		} else {
			a = 2;
			a = 3;
		}
		StaticJavaLib.assertTrue(a == 3);

	}
}
